### PR TITLE
Add ArcxAnalyticxProvider and an example CRA app

### PR DESCRIPTION
**Pull request by gtupak**
The SDK as-is was a bit of a pain to use as users need to make a custom react provider in their react apps. For this reason, I made a utility `ArcxAnalyticsProvider` that does that for you. It requires the `apiKey` and can optionally take a `config` object.

The provider initializes the SDK and exposes a hook `useArcxAnalytics()` that children of the provider can use.

The tests for the provider will be in a subsequent PR because of some configuration changes regarding the test framework and I don't want to make this PR too big.

In addition, I've added an example Create React App showcasing how to use the provider, with a console-like view showing what events have been fired. I tested this app using `yarn link`, but once the npm package is updated we'll be able to test it using the official npm package.

image

### CODE REVIEW

**Review by kQuintal — CHANGES_REQUESTED** [review]
Love this 🚀  The Provider Component is super proffesional, and example app is a great idea.

Two things:
* I can't get the example app to compile tho? From a fresh clone on this branch, might be me?
image

* Also would be neat to update tab metadata:
image

**Comment by gtupak** [comment]
This is because the npm package is not yet published. The way to make it work on this branch is to:
1. At the root folder, run `yarn link`, then `yarn build`
2. Go to the example repo and run `yarn link "@arcxmoney/analytics"

Then try again. The idea however is to publish this package and make the example repo pull the package from npm directly.

File: package.json [hunk 1]
```diff
@@ -43,5 +50,9 @@
     "clean": "rm -rf dist",
     "lint": "eslint '{src,test}/**/*.ts'",
     "test": "mocha -r ts-node/register 'test/**/*.test.ts'"
+  },
+  "peerDependencies": {
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
```
Comment (gtupak) [inline package.json:56]: Not sure if we need react-dom here

Comment (gtupak) [reply package.json:56]: Test react 17 and update the dependency range

Comment (gtupak) [reply package.json]: tested versions all the way down to 16 and the build succeeded

File: example/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Usage of ARCx Analyticx React Provider
+
+If you have a React app, one easy way to integrate the SDK is to simply add the `ArcxAnalyticsProvider` at the root of your application:
+
+You must pass an `API_KEY` as specified in the [SDK's documentation](https://github.com/arcxmoney/analytics-sdk) and, optionally, a `config` object. This is the same `config` as described [here](https://github.com/arcxmoney/analytics-sdk#init).
+
+```jsx
+return (
+    <ArcxAnalyticsProvider apiKey={API_KEY} config={config}>
+        <div>
```
Comment (xitronix) [inline example/README.md:12]: I suppose <div> is redundant here

File: example/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  "scripts": {
+    "start": "rm -rf ../node_modules/{react,react-dom} && react-scripts start",
+    "build": "react-scripts build",
+    "eject": "react-scripts eject"
+  },
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
```
Comment (xitronix) [inline example/package.json:27]: Why do we need react-app/jest here?

File: example/public/robots.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
```
Comment (xitronix) [inline example/public/robots.txt:2]: is it necessary for local example?

File: package.json [hunk 5]
```diff
@@ -16,18 +16,24 @@
     "@restless/sanitizers": "^0.2.5"
   },
   "devDependencies": {
+    "@babel/core": "^7.0.0-0",
+    "@babel/plugin-syntax-flow": "^7.14.5",
+    "@babel/plugin-transform-react-jsx": "^7.14.9",
```
Comment (xitronix) [inline package.json:21]: why do we need babel?

File: example/src/react-app-env.d.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (xitronix) [inline example/src/react-app-env.d.ts:1]: ?

**Review by xitronix — APPROVED** [review]
Nothing critical, but address the comments at your discretion

File: example/package.json (hunk 3 above)
Comment (gtupak) [reply example/package.json:27]: It came with CRA, but forgot to remove it :)

File: example/public/robots.txt (hunk 4 above)
Comment (gtupak) [reply example/public/robots.txt:2]: It came with CRA, I'll leave it there in case we want to deploy the example project

File: package.json (hunk 5 above)
Comment (gtupak) [reply package.json:21]: peer dependencies requirement

File: example/src/react-app-env.d.ts (hunk 6 above)
Comment (gtupak) [reply example/src/react-app-env.d.ts:1]: https://stackoverflow.com/questions/67262914/what-is-the-react-app-env-d-ts-in-a-react-typescript-project-for

File: example/public/robots.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (gtupak) [reply example/public/robots.txt:2]: answered above

File: package.json (hunk 5 above)
Comment (xitronix) [reply package.json:21]: From which package? :)

**Review by kQuintal — APPROVED** [review]
Discussed via dm's

File: package.json (hunk 5 above)
Comment (gtupak) [reply package.json:21]: `eslint-config-react-app`

Comment (gtupak) [reply package.json:21]: .. but now I realized we don't use it in this scope, so removed it! :) good catch!

**Pull request merged**